File: Framework/Framework.Klowd/Packages/framework/fw/modules/mvc/engine/scope.js
// ============================================================================
// Project: Framework
// Name/Class: mvc.engine.scope
// Created On: 28/Mar/2016
// Author: João Carreiro (joaopaulocarreiro@gmail.com)
// Company: Coop4Creativity
// ============================================================================

'use strict';
fw.module('mvc.engine').service('scope', 'core.util', function ($util) {

    //
    // Library name.
    //

    var __LIB = 'mvc.engine.scope';

    //
    // Get the root scope for app.
    //

    var _root = function () {

        var scope = {

            //
            // Parent scope value.
            //

            parent: null
        };

        return scope;
    };

    //
    // Get a new scope object based on am existing scope.
    //

    var _new = function (parent, container, fragment) {

        var scope = {

            //
            // Parent scope value.
            //

            parent: parent,

            //
            // Container for fragment.
            //

            container: container,

            //
            // Fragment reference for scope.
            //

            fragment: fragment,

            //
            // Processed fragment model.
            //

            model: null,

            //
            // Processed and original view.
            //

            view: null,

            //
            // List of bindings definition.
            //

            binding: null,

            //
            // Tree of components.
            //

            tree: null,

            //
            // Generated html
            //

            html: null
        };

        return scope;
    }

    //
    // API
    //

    return {
        'root': _root,
        'new': _new
    };
});

File: Framework/Framework.App/Services/Scope.cs
// ============================================================================
// Project: Framework
// Name/Class: SrvScope
// Author: João Carreiro (joao.carreiro@coop4creativity.com)
// Create date: 26/Nov/2015
// Company: Coop4Creativity
// Description: Scope service implementation..
// ============================================================================

using Framework.App.Runtime;
using Framework.Core.Api;
using Framework.Core.Patterns;

namespace Framework.App.Services
{
    public class Scope : ACommon, IScope
    {
        //
        // PROPERTIES
        //

        public IContainer Hub
        {
            get { return Manager.Container; }
        }

        //
        // Mrthod to return a new scope based on 
        // the current scope.
        // @return A new derived runtime scope.
        //

        public IScope New()
        {
            return this;
        }
    }
}
